wlauto/workloads/googleslides/__init__.py

```diff
@@ -0,0 +1,147 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import os.path as path
+import re
+from wlauto import AndroidUiAutoBenchmark, Parameter
+from wlauto.utils.types import list_of_strings
+
+
+def log_method(workload, name):
+    workload.logger.info('===== {}() ======'.format(name))
+
+class GoogleSlides(AndroidUiAutoBenchmark):
+
+    name = 'googleslides'
+    package = 'com.google.android.apps.docs.editors.slides'
+    description = 'Creates a Google Slides presentation with some commonly used features'
+    activity = ''
+
+    # Views for FPS instrumentation
+    view = [
+        package + '/com.google.android.apps.docs.quickoffice.filepicker.FilePickerActivity',
+        package + '/com.google.android.apps.docs.editors.shared.filepicker.FilePickerActivity',
+        package + '/com.google.android.apps.docs.quickoffice.filepicker.LocalSaveAsActivity',
+        package + '/com.qo.android.quickpoint.Quickpoint',
+        package + '/com.google.android.apps.docs.app.DocsPreferencesActivity',
+        package + '/com.google.android.apps.docs.app.DocListActivity',
+        package + '/com.google.android.apps.docs.welcome.warmwelcome.TrackingWelcomeActivity',
+        package + '/com.google.android.apps.docs.app.NewMainProxyActivity',
+    ]
+
+    parameters = [
+        Parameter('dumpsys_enabled', kind=bool, default=True,
+                  description='''
+                  If ``True``, dumpsys captures will be carried out during the test run.
+                  The output is piped to log files which are then pulled from the phone.
+                  '''),
+        Parameter('local_file', kind=str,
+                  description='''
+                  If specified, the workload will push the PowerPoint file to be used for
+                  testing on the device. Otherwise, a file will be created inside the app.
+                  '''),
+        Parameter('slide_count', kind=int, default=5,
+                  description='''
+                  Number of slides in aforementioned local file. Determines number of
+                  swipe actions when playing slide show.
+                  '''),
+    ]
+
+    instrumentation_log = '{}_instrumentation.log'.format(name)
+
+    def __init__(self, device, **kwargs):
+        super(GoogleSlides, self).__init__(device, **kwargs)
+        self.run_timeout = 300
+        self.output_file = path.join(self.device.working_directory, self.instrumentation_log)
+        self.local_dir = self.dependencies_directory
+        # Android downloads folder
+        self.device_dir = path.join(self.device.working_directory, '..', 'Download')
```

Comment:
In the 4 devices I was able to get my hands on (if only momentarily) to test on, this seemed to guarantee that the file will reliably found — `Download` is the default folder for the `Open` action in Slides, and the search function only includes online files, or offline files previously opened by the app.

Otherwise, pushing it to WA's working directory meant trying to navigate up/back in the hierarchy to try and find it. This was not as reliable as the appearance of the file picker differed by device.

I'd be happy to tweak the implementation if we find that the `Download` folder is in fact less reliable.